spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/hazelcast/HazelcastClientConfiguration.java

```diff
@@ -81,6 +87,33 @@ HazelcastInstance hazelcastInstance(ClientConfig config) {
 					"file:./hazelcast-client.yaml", "classpath:/hazelcast-client.yaml");
 		}
 
-	}
+		@Override
+		public ConditionOutcome getMatchOutcome(ConditionContext context, AnnotatedTypeMetadata metadata) {
+			String springHazelcastConfigProperty = "spring.hazelcast.config";
+			if (context.getEnvironment().containsProperty(springHazelcastConfigProperty)) {
+				String configLocation = context.getEnvironment().getProperty(springHazelcastConfigProperty);
+				Resource resource = context.getResourceLoader().getResource(configLocation);
+				if (resource.exists()) {
+					try {
+						validateHazelcastClientConfig(resource);
+						return ConditionOutcome.match(startConditionMessage().foundExactly("property "
+								+ springHazelcastConfigProperty));
+					} catch (Exception e) {
```

Comment:
This is catching a large amount of cases here, not only the fact that the builder was instantiated with an unexpected configuration file. 

Isn't there a less agressive way to detect the config refers to a server rather than a client?